packages/models/src/sql/log.js

```diff
@@ -0,0 +1,29 @@
+import { open, write } from 'fs';
+
+let SqlWarningFileName = 'sql_warning.txt';
+let messages /** @type {string[]} */ = [];
+let writeMessage = (msg) => messages.push(msg);
+
+process.on('exit', () => {
+  if (!messages) return;
+
+  [...new Set(messages)].forEach((msg) => console.warn(msg));
+});
+
+// eslint-disable-next-line import/prefer-default-export
+export function warn(/** @type {string} */ message) {
+  if (SqlWarningFileName) {
+    open(SqlWarningFileName, 'w', (err, fd) => {
+      if (err || !fd) return;
```

Comment:
This will silently drop messages. Either show a warning or fallback to console.

Reply:
It shouldn't drop messages, because line 28 will add the message to the `messages` array, and they will all be dumped out on exit.

Reply:
Turns out `@appland/models` is not allowed to have dependencies on Node.js, so I removed it.

Reply:
This package will simply report errors to a function that the client can optionally provide.

Reply:
> It shouldn't drop messages, because line 28 will add the message to the `messages` array, and they will all be dumped out on exit.

Ah, you're right I missed the lambda :)